src/devices/src/virtio/persist.rs

```diff
@@ -95,13 +109,122 @@ impl VirtioDeviceState {
     }
 }
 
+#[derive(Versionize)]
+// Clippy is complaining now because of the large size difference between variants.
+// Once we add Net state, the clippy exception can be removed.
+#[allow(clippy::large_enum_variant)]
+enum DeviceTypeState {
+    Block(BlockState),
+    // TODO: add net state when available
+    Net,
+    Vsock(VsockState),
+}
+
+#[derive(Versionize)]
+pub struct MmioTransportState {
+    // The register where feature bits are stored.
+    features_select: u32,
+    // The register where features page is selected.
+    acked_features_select: u32,
+    queue_select: u32,
+    device_status: u32,
+    config_generation: u32,
+    device_state: DeviceTypeState,
```

Comment:
I would not bundle the actual device state inside the MmioTransportState. We should decouple these 2 things and the MmioTransport state should be serialized separately and the constructor struct should contain Arc<Mutex<dyn VirtioDevice>> that is constructed at the top level.

Actually we are restoring a device from lego parts that can be bundled together like this:
```
struct GenericDeviceState {
     transport_state: MmioTransportState,
     device_state: VsockState,
}
```

Reply:
> Actually we are restoring a device from lego parts that can be bundled together like this:

You mean:
```rust
struct GenericDeviceState {
     transport_state: MmioTransportState,
     device_state: SomeEnumWithSpecificDeviceVariants,
}
```
right?